### PR TITLE
Add gallery example for grdview

### PR DESCRIPTION
**Pull request by liamtoney**
**Description of proposed changes**

Add an example use of [`grdview()`](https://www.pygmt.org/latest/api/generated/pygmt.Figure.grdview.html#pygmt.Figure.grdview) to the [gallery](https://www.pygmt.org/latest/gallery/index.html). Serves as an example for https://github.com/GenericMappingTools/pygmt/issues/433, as well as a refresher for me. 😝 

**Reminders**

- [x] Run `make format` and `make check` to make sure the code follows the style guide.
- [ ] Add tests for new features or tests that would have caught the bug that you're fixing.
- [ ] Add new public functions/methods/classes to `doc/api/index.rst`.
- [ ] Write detailed docstrings for all functions/methods.
- [ ] If adding new functionality, add an example to docstrings or tutorials.

### CODE REVIEW

**Comment by liamtoney** [comment]
Two minor comments on this:

1. I didn't actually preview the ReST of the docstring. Do we have the math extension? Otherwise the `:math:` directive won't render properly.

2. I tried to illuminate the surface, but `I='+'` gave a flat image with shading rather than an illuminated 3-D surface.

File: examples/gallery/grid/grdview_surface.py [hunk 1]
```diff
@@ -0,0 +1,49 @@
...
+axis scaling.
+"""
+
+import pygmt
+import numpy as np
+import xarray
+
+# Define an interesting function of two variables, see:
+# https://en.wikipedia.org/wiki/Ackley_function
+def ackley(x, y):
```
[the remark this answers is not in the document]
Comment (liamtoney) [reply examples/gallery/grid/grdview_surface.py:21]: Added a line

**Comment by weiji14** [comment]
> 1. I didn't actually preview the ReST of the docstring. Do we have the math extension? Otherwise the `:math:` directive won't render properly.

It might be tough for Scipy Sprint users to preview the documentation locally, unless they've set up a PyGMT environment. Luckily we've got #344, which reminds me, I'll send you an email to invite you as an owner on the [Vercel](https://vercel.com) gmt team, since you're part of @GenericMappingTools/python now. Anyways, you can preview things at https://pygmt-git-fork-liamtoney-grdview-gallery.gmt.now.sh/gallery/grid/grdview_surface.html#plotting-a-surface.

> 2. I tried to illuminate the surface, but `I='+'` gave a flat image with shading rather than an illuminated 3-D surface.

Haven't tried illumination with `grdview` before, maybe try with debug mode on using `V="d"`?

**Comment by liamtoney** [comment]
> Anyways, you can preview things at https://pygmt-git-fork-liamtoney-grdview-gallery.gmt.now.sh/gallery/grid/grdview_surface.html#plotting-a-surface.

That is really nifty!

> Haven't tried illumination with grdview before, maybe try with debug mode on using V="d"?

Didn't see anything obvious in the console output. Will have to table for later, it looks alright as is!

**Comment by liamtoney** [comment]
> Didn't see anything obvious in the console output. Will have to table for later, it looks alright as is!

I was able to re-create the code in pure GMT, and the shading performs as expected, so it's a PyGMT grid I/O issue I think. I'll make an issue.

File: examples/gallery/grid/grdview_surface.py [hunk 2]
```diff
@@ -0,0 +1,50 @@
...
+``y`` defined. Note that the ``perspective`` argument here controls the azimuth and
+elevation angle of the view. We provide a list of two arguments to ``frame`` — the
+second argument, prepended with ``"z"``, specifies the :math:`z`-axis frame attributes.
+Specifying the same scale for the ``projection`` and ``zcale`` arguments ensures equal
+axis scaling.
+"""
+
+import pygmt
+import numpy as np
+import xarray
```
Comment (seisman) [inline examples/gallery/grid/grdview_surface.py:16]:
```suggestion
import xarray as xr
```
This is the most commonly used way to import the xarray package.

**Comment by seisman** [comment]
@weiji14 is the one implementing the `grdview()` functions. Any comments?

**Comment by liamtoney** [comment]
Whoops. I saw your approval and merged. Apologies if that was premature.

**Comment by seisman** [comment]
It's fine. We can fix it later if @weiji14 wants more.

**Comment by weiji14** [comment]
Oops, missed it by a few minutes. I actually really like this Ackley function example, definitely better than anything I could have come up with!

Aside from a few extra inline comments, I think it's fine as is. We can add them in when the shading works in GMT 6.1 :wink:

**Comment by liamtoney** [comment]
> We can add them in when the shading works in GMT 6.1 😉

Yes the [shaded version](https://forum.generic-mapping-tools.org/t/surface-plot-of-ackley-function/601
) looks quite nice so I'd like to update this example once we get to 6.1!

**Pull request merged**